### PR TITLE
only b2c in swagger

### PR DESCRIPTION
**Pull request by dvoet**
Ticket: https://broadworkbench.atlassian.net/browse/TOAZ-349

update to latest version of workbench-libs oauth2 and adopt open api `securitySchemes`:
```
    oidc:
      type: openIdConnect
      openIdConnectUrl: OPEN_ID_CONNECT_URL
      x-tokenName: id_token
    oidc_google_billing_scope:
      type: openIdConnect
      openIdConnectUrl: OPEN_ID_CONNECT_URL_WITH_GOOGLE_BILLING_SCOPE
      x-tokenName: id_token
```

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/resources/swagger/api-docs.yaml [hunk 1]
```diff
@@ -9771,22 +9770,22 @@ components:
...
             profile: profile authorization
-            https://www.googleapis.com/auth/cloud-billing: GCS billing
-    oidc:
+    oidc_google_billing_scope:
       type: oauth2
       flows:
         authorizationCode:
-          authorizationUrl: /oauth2/authorize
-          tokenUrl: /oauth2/token
+          authorizationUrl: /oauth2/authorize?p=B2C_PROFILE_WITH_GOOGLE_BILLING_SCOPE
```
Comment (rtitle) [inline src/main/resources/swagger/api-docs.yaml]: As mentioned in the other PR, I'm a little curious what happens if `additionalQueryStringParams` is used.

https://swagger.io/docs/open-source-tools/swagger-ui/usage/oauth2/

File: local-dev/templates/firecloud-orchestration.conf.ctmpl [hunk 2]
```diff
@@ -64,7 +64,7 @@ akka {
 auth {
 	authorityEndpoint = "https://terra{{ $b2cTenant }}b2c.b2clogin.com/terra{{ $b2cTenant }}b2c.onmicrosoft.com/v2.0?p=b2c_1a_signup_signin_{{ $b2cEnv }}"
 	oidcClientId = "{{ $b2cAppId }}"
-	legacyGoogleClientId = "{{ $refreshTokenCredential.web.client_id }}"
+	b2cProfileWithGoogleBillingScope = "b2c_1a_signup_signin_billing_{{ $b2cEnv }}"
```
Comment (rtitle) [inline local-dev/templates/firecloud-orchestration.conf.ctmpl]: Nice, I guess prior to this Rawls and Orch swagger don't have a way to request GCP billing scopes via B2C..

Comment (rtitle) [reply local-dev/templates/firecloud-orchestration.conf.ctmpl]: Does this need some helmfile update as well?

**Review by rtitle — APPROVED** [review]

**Review by davidangb — APPROVED** [review]
:+1:. did not try it myself, but checked out your bee.

Is there any way to get the swagger-ui to not show both `oidc (OAuth2, authorization_code with PKCE)` and `oidc (OAuth2, implicit)`?

**Comment by dvoet** [comment]
> Is there any way to get the swagger-ui to not show both `oidc (OAuth2, authorization_code with PKCE)` and `oidc (OAuth2, implicit)`?

I did not see one but did not look that hard, will look again

**Pull request merged**